### PR TITLE
[WFCORE-2017] Don't enable the access=identity service if running in embedded admin only mode.

### PR DESCRIPTION
**Pull request by darranl**
https://issues.jboss.org/browse/WFCORE-2017

### CODE REVIEW

**Comment by bstansberry** [comment]
Why is this service never appropriate in an embedded server?

**Comment by bstansberry** [comment]
Never mind; I misread the change.

File: domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java [hunk 1]
```diff
@@ -170,7 +172,7 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
 
         @Override
         protected boolean requiresRuntime(OperationContext context) {
-            return true;
+            return context.getProcessType() != ProcessType.EMBEDDED_SERVER || context.getRunningMode() != RunningMode.ADMIN_ONLY;
```
Comment (bstansberry) [inline domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java]: This needs to account for ProcessType.EMBEDDED_HOST_CONTROLLER as well.

I'm not sure on this but perhaps ProcessType could expose and isEmbedded() property to help with keeping this particular kind of discrimination consistent. My only hesitation is whether that will always have a single value for all process types. For example, for SELF_CONTAINED it's now false but maybe someday there would be a flavor of that kind of thing where it is ambiguous.

Comment (bstansberry) [inline domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java]: I'll make ^^^ part of a formal "Request changes" review.

Comment (darranl) [inline domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java]: The HTTP management interfaces don't make that check - I have tried to make sure this matches those so when they come up this comes up.

Comment (darranl) [inline domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java]: This change matches the check for the management interfaces, I have tried to match the interface as it is important if they come up this comes up.

Comment (darranl) [inline domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java]: Never mind - I see it in the other add handler.

Comment (darranl) [inline domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java]: Never mind - I see it in the other add handler.

**Pull request merged**